### ISSUE TITLE
RFC: Send emails based on a future date

### ISSUE THREAD

**Issue by mikecasas** [oqtane/oqtane.framework#567]
I propose to add a new field to the Notification table called ``SendOn``. This field would allows emails to be sent either immediately or sometime in the future.

The default implementation would have a SendOn date of 'Now', which wouldn't change the current framework implementation. Developers could then add their own notifications with a SendOn future date, thereby having a simple email reminder system.

I have a PR ready, but was wondering if this would be good for a dev branch? Also, I saw a SQL script with a 1.0.1 reference, is there an 0.1 update coming soon?

**Comment by sbwalker** [comment]
In order to add a database change in an upgradeable manner it needs to be part of the next release version script

**Comment by thabaum** [comment]
does this use the scheduler to send mails at those times? or are you adding only the db field and the module handles the notification along with creating a scheduler?

**Comment by mikecasas** [comment]
@thabaum I was planning to modify the scheduler. In the ```NotificationRepository```, my change would be 
```
 return _db.Notification
                    .Where(item => item.SiteId == siteId)
                    .Where(item => item.IsDelivered == false)
                    .Where(item => item.SendOn < System.DateTime.UtcNow))
                    .Include(item => item.FromUser)
                    .Include(item => item.ToUser)
                    .ToList();

```

I have a branch named ```notification-feature`` in my repo if you want to look at it. I was waiting until after the 1.0.1 stabilization release to submit it.

**Comment by sbwalker** [comment]
This is not a modification to the Oqtane scheduler - it is a modification to the Notification scheduled job, correct?

**Comment by mikecasas** [comment]
Sorry, yes. I'm not touching the Scheduler, just the notifications.

**Comment by thabaum** [comment]
cool i will take a look at it @mikecasas thanks for the reply

**Comment by thabaum** [comment]
I have two PR's in the same area ready to go @sbwalker that may conflict since this is a related component.  Can you please take a look at these and merge when you have time if they are ok prior to merging this PR once accepted?

I put some icing on the cake on these I believe. 

#586
#585

**Comment by sbwalker** [comment]
@mikecasas please go ahead and submit this PR

**Comment by mikecasas** [comment]
@sbwalker I'll look at it tonight. I need to dust it off.

**Comment by mikecasas** [comment]
In testing the changes, it seems when a user registers on a site the ```ToEmail``` field is entered as blank. When the Notification job runs it errors out because the ToEmail field is blank.

This has nothing to do with my changes.

Line 151
https://github.com/mikecasas/oqtane.framework/blob/master/Oqtane.Server/Controllers/UserController.cs#L132-L161

Line 81
https://github.com/mikecasas/oqtane.framework/blob/master/Oqtane.Server/Infrastructure/Jobs/NotificationJob.cs#L79-L88

**Comment by sbwalker** [comment]
it looks like a bug - can you fix it as part of your PR?

**Comment by mikecasas** [comment]
I included the fix along with the enhancement in #625.
